stan/math/opencl/prim/bernoulli_logit_glm_lpmf.hpp

```diff
@@ -119,7 +124,8 @@ return_type_t<T_alpha, T_beta> bernoulli_logit_glm_lpmf(
                  from_matrix_cl(x_cl));
   }
 
-  operands_and_partials<T_alpha, T_beta> ops_partials(alpha, beta);
+  operands_and_partials<decltype(alpha_ref), decltype(beta_ref)> ops_partials(
```

Comment:
`to_ref<T>` and `to_ref_if<T>` actually never return `Eigen::Ref`. They return either `plain_type_t<T>` or `const T&`, using the same mechanism as `Eigen::Ref` to decide which one. The type we put in here is the type of argument the constructor of `ops_and_partials_edge` accepts. So if we put `plain_type_t<T_alpha>` it would evaluate an expression when calling the constructor, before copying it in local variable. 

While this copy might be optimized away by the compiler, we can instead accept exact type, so there is no copy.

Reply:
Right, that makes sense, thanks